### ISSUE TITLE
Suggestion: add a flag to bypass setting zsh as default shell

### ISSUE THREAD

**Issue by left4taco** [robbyrussell/oh-my-zsh#7100]
Currently oh-my-zsh install.sh script will try to set the default shell to zsh. But sometimes this will prompt the user to enter their password. This is somewhat annoying in auto-deploying scripts. I would suggest to add a flag to `install.sh` to bypass this section. Or is there already one? Thanks. 

```sh
  # If this user's login shell is not already "zsh", attempt to switch.
  TEST_CURRENT_SHELL=$(expr "$SHELL" : '.*/\(.*\)')
  if [ "$TEST_CURRENT_SHELL" != "zsh" ]; then
    # If this platform provides a "chsh" command (not Cygwin), do it, man!
    if hash chsh >/dev/null 2>&1; then
      printf "${BLUE}Time to change your default shell to zsh!${NORMAL}\n"
      chsh -s $(grep /zsh$ /etc/shells | tail -1)
    # Else, suggest the user do so manually.
    else
      printf "I can't change your shell automatically because this system does not have chsh.\n"
      printf "${BLUE}Please manually change your default shell to zsh!${NORMAL}\n"
    fi
  fi

```

**Comment by marshallford** [comment]
I'm going to take a look at this.

**Comment by rahul0705** [comment]
Bump for this feature, Make having an auto deployable dotfile project slightly harder.

I use ZSH when working in HPC environments but I leave my default shell to bash so that all the environmental stuff in the cluster will work correctly. This is a place where I manually launch ZSH and make sure its not my default shell.

**Comment by mcornella** [comment]
This has been done in #5169 from changes in #7266, and is now ready for testing.

If you want to help speed up the testing and merge (which should happen early next week), you can wget or curl it from the URL `https://raw.githubusercontent.com/mcornella/oh-my-zsh/refactor-installer/tools/install.sh`. Read the comments at the top of the script to know how to call it.

Thanks!

**Comment by left4taco** [comment]
@mcornella Thank you. I actually had my own solution a year ago. But I just tested your code and it works perfectly. Thanks.

**Comment by mcornella** [comment]
That's great to hear, thanks!

**Comment by mcornella** [comment]
Closed by #5169.
